libs/deal-ticket/src/components/deal-ticket/deal-ticket-stop-order.tsx

```diff
@@ -50,32 +48,40 @@ export interface StopOrderProps {
   submit: (order: StopOrdersSubmission) => void;
 }
 
-const defaultValues: Partial<StopOrderFormValues> = {
+const defaultValues: StopOrderFormValues = {
   type: Schema.OrderType.TYPE_LIMIT,
   side: Schema.Side.SIDE_BUY,
   timeInForce: Schema.OrderTimeInForce.TIME_IN_FORCE_FOK,
   triggerType: 'price',
   triggerDirection:
     Schema.StopOrderTriggerDirection.TRIGGER_DIRECTION_RISES_ABOVE,
+  expire: false,
   expiryStrategy: Schema.StopOrderExpiryStrategy.EXPIRY_STRATEGY_SUBMIT,
   size: '0',
 };
 
-const stopSubmit: FormEventHandler = (e) => e.preventDefault();
-
 export const StopOrder = ({ market, marketPrice, submit }: StopOrderProps) => {
   const { pubKey, isReadOnly } = useVegaWallet();
-  const setDealTicketType = useDealTicketTypeStore((state) => state.set);
-  const [, updateOrder] = useOrder(market.id);
-  const updateStoredFormValues = useStopOrderFormValues(
-    (state) => state.update
+  const setType = useDealTicketFormValues((state) => state.setType);
+  const dealTicketType = useDealTicketFormValues((state) =>
+    state.formValues[market.id]?.type === DealTicketType.StopMarket
```

Comment:
What if the stored ticket type is normal limit or normal market? This doesn't look right to me?